### PR TITLE
Code Review Jacob Farrish

### PR DESCRIPTION
**Pull request by jakeFarrish**
This is an app to learn more about mental health issues and gain some information that may help a user cope and deal with the issue and find resources that can help them address the issues. The basic functionality is fleshed out but no content so far, there seems to be dto's for each main element, but no logic applied which is understandable at this phase. UI is very simple set up so far, so i added some structure to the resources folders and updated the start.html to pull from a thymeleaf layout page as well as added a main.css and index.js files and a web mvc unit test.

**Summary of Changes**

Added web content unit test (testing for the welcome to mental health app text), added folder for static UI resources (this is basic structure for a springboot project, all thymeleaf static resources can be linked when placed in a static resources folder) 

Added empty css.main and empty index.js (just in case you need to add custom styles or js) 
added bootstrap.min.css (there was an error that bootstrap.min was missing, also downloaded css and js libraries that were missing) 

Added layout page and linked start page to layout as well as defined content fragments in start.html and inserted thymeleaf dependency into pom.xml and updated dependency versions to current release. The layout page is a nice way to structure the ui of you application to a standard layout that can be modified with fragment areas using thymeleaf

Created the initial links between start and layout.html, I would advise structuring the nav bar in the layout outside of the content fragment though as the nav is a static element that shouldn't change too much from page to page.  

Program Available on Github?
Yes

Program Understandable?
Yes, straightforward concept.

Compile without errors?
Yes, no errors on compile / run.

What I learned:
 
I learned how to create a pull request on Github
I learned how to update maven project dependencies in IntelliJ and to download maven resources from the maven command menu
I learned how to link an external stylesheet and a local stylesheet to a page and how the style rules apply with two stylesheets.

My Commits in Labor Helper:

[Adding WebMvc weblayer content test](https://github.com/ccbrowndev/labor-helper/commit/4121f253f59a13f12d302f9b485e6adc24401eba)

[Created initial UI, added thymeleaf layout structure and dependencies, added static resources](https://github.com/ccbrowndev/labor-helper/commit/609b50652473564140347d09b2641399ae66557a)

[Modified layout styles and added unstyled login/registration page
](https://github.com/ccbrowndev/labor-helper/commit/9186b3a8a3211a19716ed0c577183297fd4677c0)

### CODE REVIEW

File: src/test/java/com/mentalhealthapi/mentalhealthapi/MentalHealthApiWebTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.util.AssertionErrors.assertEquals;
+
+@SpringBootTest
+class LaborHelperApplicationTests {
+
+    @Test
+    void contextLoads() {
+    }
```
Comment (discospiff) [inline src/test/java/com/mentalhealthapi/mentalhealthapi/MentalHealthApiWebTest.java:23]: This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

**Comment by discospiff** [comment]
Layout is a good idea.

Avoid adding empty files... until you have something to put in them.  Unused code is clutter, which is technical debt.

File: src/test/java/com/mentalhealthapi/mentalhealthapi/MentalHealthApiWebTest.java (hunk 1 above)
Comment (ghost) [reply src/test/java/com/mentalhealthapi/mentalhealthapi/MentalHealthApiWebTest.java:23]: We removed this some time ago

**Comment by sash4spur** [comment]
Overall, I believe that the layout file would make our web app more scalable and easier to write. I will heavily consider this suggestion, but will not merge due to merge conflicts. We have had major file rehauls, deletions, etc.

Thank you

**Pull request closed without merging**